internal/service/common/utils/utils.go

```diff
@@ -115,3 +125,52 @@ func GetDBTagsFromStructFields[T db.Model](s T, fields ...string) DBTag {
 
 	return tags
 }
+
+// CompareObjects compares two objects and returns the tags for fields having differing values.  Any field names listed
+// in `excluded` are ignored.
+func CompareObjects[T db.Model](a, b T, excluded ...string) DBTag {
```

Comment:
I can see where this function is used in subsequent commits. I have many concerns/questions there.
I think we are abusing of reflect and generics. I see that you use it to update only some fields but I wonder if it is worth the trouble. As far as I know, postgres is a mvcc db which means an update is an insert-delete underneath. I know, there is some gain by not sending everything, but is it really too much?  How often do you expect to have changes.

And if performance is a concern here, then I see you are using the slowest approach, updating an object at a time. CopyFrom is the fastest, then single query with multiple rows, after that batch (multiiple queries inside) and finally one row at a time. https://github.com/jackc/pgx/discussions/1545

Also, postgres has an upsert verb that can be used instead of checking in code if an insert or update is needed.

I'm almost done reviewing the first 3 commits of this PR. I suggest you split this pr in 2, one with the first 3 commits so that we can merge them today. The last 3 will require more time. If you do that, please remove this function from the "CNF-15570: Add support for updating only some fields" commit

Reply:
I don't anticipate that the volume of updates will be high.  I was following best practices of only updating necessary fields.  This is what most ORMs do to avoid overloading the database with updating unnecessary fields.   We can certainly remove that, but from a debug point of view it makes it difficult to tell why something is being updated if everything is being written on all updates.

Reply:
I should qualify my comment about the volume of updates.  As the code is written now every record will get updated to set the generation id to track stale records.   I could switch to a different approach to manage that and we wouldn't need as many updates.

Reply:
@mlguerrero12 as per your request I have split the PR.  The other one with the top 3 commits (plus a new one for the subscription notification handling) is over [here](https://github.com/openshift-kni/oran-o2ims/pull/384).